### PR TITLE
Allow refreshing multiple locks at once

### PR DESCRIPTION
**Pull request by p-offtermatt**
## Description

Closes: #123 
*Add a description of the changes that this PR introduces and the files that
are the most critical to review.*

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [x] Targeted the correct branch
* [x] Included the necessary unit tests
* [x] Added/adjusted the necessary [interchain tests](./test/interchain/)
* [x] Added a changelog entry in `.changelog`
* [x] Compiled the contracts by using `make compile` and included content of the *artifacts* directory into the PR
* [x] Regenerated front-end schema by using `make schema` and included generated files into the PR
* [x] Updated the relevant documentation or specification
* [x] Reviewed "Files changed" and left comments if necessary
* [x] Confirmed all CI checks have passed

### CODE REVIEW

File: contracts/hydro/src/testing.rs [hunk 1]
```diff
@@ -1926,3 +1926,126 @@ pub fn pilot_round_lock_duration_test() {
...
+fn test_refresh_multiple_locks() {
+    let grpc_query = denom_trace_grpc_query_mock(
+        "transfer/channel-0".to_string(),
+        HashMap::from([(IBC_DENOM_1.to_string(), VALIDATOR_1_LST_DENOM_1.to_string())]),
+    );
+    let (mut deps, mut env) = (mock_dependencies(grpc_query), mock_env());
+    let info = get_message_info(&deps.api, "addr0000", &[]);
+
+    // Define test cases
+    let test_cases = vec![
```
Comment (dusan-maksimovic) [inline contracts/hydro/src/testing.rs:1954]: Add another test case that tries to extend lock that belongs to different user?

File: contracts/hydro/src/contract.rs [hunk 2]
```diff
@@ -312,36 +314,66 @@ fn refresh_lock_duration(
...
+            new_lock_end.to_string(),
+        );
+    }
 
-    // check that the new lock_end_time is later than the old lock_end_time
+    Ok(response)
+}
+
+fn refresh_single_lock(
+    deps: DepsMut<'_, NeutronQuery>,
```
[the remark this answers is not in the document]
Comment (p-offtermatt) [reply contracts/hydro/src/contract.rs]: Great point, thanks. I think I did it the way you intended - please have a look!

**Comment by p-offtermatt** [comment]
Thanks, both are done!

**Review by dusan-maksimovic — APPROVED** [review]
LGTM!

**Pull request merged**